cpp/daal/include/algorithms/boosting/adaboost_predict.h

```diff
@@ -131,7 +131,11 @@ class DAAL_EXPORT Batch : public classifier::prediction::Batch
      */
     Batch(const Batch<algorithmFPType, method> & other);
 
-    ~Batch() { delete _par; }
+    /**
+     * Destructor.
+     * Releases dynamically allocated resources held by the object.
+     */
+    virtual ~Batch() { delete _par; }
```

Comment:
'virtual' is not needed here and in other similar places, as the base class already declares the destructor as 'virtual'.
```suggestion
    ~Batch() { delete _par; }
```